composer/trainer/trainer.py

```diff
@@ -565,12 +635,89 @@ def eval_subset_num_batches(self):
     def eval_subset_num_batches(self, eval_subset_num_batches: Optional[int] = None):
         self._eval_subset_num_batches = eval_subset_num_batches
 
-    def fit(self):
-        """Train and evaluate the model on the provided data."""
+    def fit(self, num_batches: Optional[int] = None, num_epochs: Optional[int] = None):
```

Comment:
I like that we're moving some params to fit, but I'm conflicted about `num_batches`. It doesn't tell me anything about the amount of information I'm feeding to the network. An `epochs` is useful as it represent a full pass through the dataset, `train_fraction` (training on a subset of the dataset) could also be useful for debugging, learning curves, and active learning situations. I know large NLP models like tokens so perhaps @moinnadeem some opinions about this?

Reply:
`num_batches` and `num_epochs` will be replaced with a `duration` argument once #146 is implemented. That will support tokens and a fraction of the total training duration.

Reply:
(I'd prefer not to add that logic here as #146 is next on my agenda)